src/main/java/io/cdap/plugin/successfactors/common/util/ExceptionParser.java

```diff
@@ -0,0 +1,173 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.SocketTimeoutException;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Collectors;
+
+/**
+ * This {@code ExceptionParser} parse and forms the relevant error messages.
+ */
+public class ExceptionParser {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ExceptionParser.class);
+
+  public static final int NO_VERSION_FOUND = 1;
+  public static final int INVALID_VERSION_FOUND = 2;
+
+  private static final Gson GSON = new Gson();
+
+  private ExceptionParser() {
+  }
+
+  /**
+   * Checks the {@code SuccessFactorsResponseContainer} status code and build appropriate error message
+   *
+   * @param msg               stage wise error message for example the caller is
+   *                          - failure while testing OData URL
+   *                          - failure while reading metadata
+   *                          - failure while reading total available record count
+   *                          - failure while reading actual data
+   * @param responseContainer {@code SuccessFactorsResponseContainer} contains the http response details after calling
+   *                          SAP SuccessFactors service
+   * @throws SuccessFactorsServiceException in case of any error scenario, it prepares and throw this exception.
+   */
+  public static void checkAndThrowException(String msg, SuccessFactorsResponseContainer responseContainer)
+    throws SuccessFactorsServiceException {
+
+    String failureMessage = msg;
+    SuccessFactorsError error = null;
+
+    if (responseContainer.getHttpStatusCode() == HttpURLConnection.HTTP_UNAUTHORIZED) {
+      failureMessage += ResourceConstants.ROOT_CAUSE_LOG.getMsgForKey()
+        .concat(ResourceConstants.ERR_INVALID_CREDENTIAL.getMsgForKey());
+
+      throw new SuccessFactorsServiceException(failureMessage, responseContainer.getHttpStatusCode());
+    } else if (responseContainer.getHttpStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+      failureMessage += ResourceConstants.ROOT_CAUSE_LOG.getMsgForKey()
+        .concat(ResourceConstants.ERR_INVALID_ENTITY_NAME.getMsgForKey());
+
+      throw new SuccessFactorsServiceException(failureMessage, responseContainer.getHttpStatusCode());
+    } else if (responseContainer.getHttpStatusCode() != HttpURLConnection.HTTP_OK) {
+      String rawResponseString = "";
+      InputStream rawStream = responseContainer.getResponseStream();
+      if (rawStream != null) {
+        try (BufferedReader bufferedReader = new BufferedReader(
+          new InputStreamReader(rawStream, StandardCharsets.UTF_8))) {
+
+          rawResponseString = bufferedReader.lines().collect(Collectors.joining(" "));
+          if (rawResponseString.contains("Refer to https")) {
```

Comment:
why are you doing this? please add comments

Reply:
Sometimes the error response returned by OData v2 API contains hyperlinks to SAP docs which leads to a page not found.
Comment added 444c10c0add0efdb6799cd258c03b10d1f479f5e

Reply:
![image](https://user-images.githubusercontent.com/89522746/170472012-3a622262-6843-413e-abdd-65820a2c4a18.png)

https://help.sap.com/viewer/d599f15995d348a1b45ba5603e2aba9b/PRODUCTION/en-US